src/adapters/scene_direction.js

```diff
@@ -189,9 +218,9 @@ export default class SceneDirection {
   }
 
   displayLabels(routes, vehicle) {
-    const labelPositions = getLabelPositions(routes.map(route => route.geometry));
+    this.labelPositions = getLabelPositions(routes.map(route => route.geometry));
```

Comment:
Is there any unwanted effect if we use `getLabelsBbbox` here to compute `labelPositions` ?

Reply:
It would be possible yes, but I think it would make the code more complex, having to compute the routes bbox here also or in advance and store it.
`fit_map` can happen outside of `displayRoutes`, so I took the same approach as before to store the mininum state and compute things on the fly. But I agree this file is quite a mess and having to store `labelPositions` is unsatisfying.

Reply:
ok, in this case, could we avoid `this.labelPositions`, by calling `getLabelPositions` where needed instead (in `getAllRoutesBBox` ?

Reply:
If it's quite straightforward go fot it :)

Reply:
Nothing is straightforward here, but I did it :)